src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/PropertySetAnalysis/PropertySetAnalysis.cs

```diff
@@ -139,40 +153,86 @@ private PropertySetAnalysis(PropertySetAnalysisDomain analysisDomain, PropertySe
+                    pessimisticAnalysis);
+                if (enclosingResult == null)
                 {
                     continue;
                 }
 
-                if (allResults == null)
+                // Also look at local functions and lambdas that weren't visited via interprocedural analysis.
+                foreach (IMethodSymbol localFunctionSymbol in enclosingControlFlowGraph.LocalFunctions)
+                {
+                    if (!enclosingResult.VisitedLocalFunctions.Contains(localFunctionSymbol))
+                    {
+                        cfgs.Add(
+                            (enclosingControlFlowGraph.GetLocalFunctionControlFlowGraph(localFunctionSymbol),
+                             localFunctionSymbol));
+                    }
+                }
+
+                foreach (IFlowAnonymousFunctionOperation flowAnonymousFunctionOperation in
+                    enclosingControlFlowGraph.DescendantOperations<IFlowAnonymousFunctionOperation>(
+                        OperationKind.FlowAnonymousFunction))
                 {
-                    allResults = PooledDictionary<(Location Location, IMethodSymbol Method), HazardousUsageEvaluationResult>.GetInstance();
+                    if (!enclosingResult.VisitedLambdas.Contains(flowAnonymousFunctionOperation))
+                    {
+                        cfgs.Add(
+                            (enclosingControlFlowGraph.GetAnonymousFunctionControlFlowGraph(flowAnonymousFunctionOperation),
+                             flowAnonymousFunctionOperation.Symbol));
+                    }
                 }
 
-                foreach (KeyValuePair<(Location Location, IMethodSymbol Method), HazardousUsageEvaluationResult> kvp
-                    in dfaResult)
+                foreach ((ControlFlowGraph cfg, ISymbol owningSymbol) in cfgs)
                 {
-                    if (allResults.TryGetValue(kvp.Key, out HazardousUsageEvaluationResult existingValue))
+                    PropertySetAnalysisResult dfaResult =
+                        cfg == enclosingControlFlowGraph
+                        ? enclosingResult
+                        : PropertySetAnalysis.GetOrComputeResult(
+                            cfg,
+                            compilation,
+                            owningSymbol,
+                            typeToTrackMetadataName,
+                            constructorMapper,
+                            propertyMappers,
+                            hazardousUsageEvaluators,
+                            interproceduralAnalysisConfig,
+                            pessimisticAnalysis);
+                    if (dfaResult == null || dfaResult.HazardousUsages.IsEmpty)
                     {
-                        allResults[kvp.Key] = PropertySetAnalysis.MergeHazardousUsageEvaluationResult(existingValue, kvp.Value);
+                        continue;
                     }
-                    else
+
+                    if (allResults == null)
```

Comment:
You can probably simplify a bit by not adding `enclosingControlFlowGraph` to `cfgs`, but directly add it to `allResults` upfront when you compute dfa result for `enclosingControlFlowGraph`.

Reply:
Yeah, was just being lazy about thinking about refactoring.  Removed the List entirely.